### PR TITLE
wip: make cli main testable

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

File: styx-cli/src/main/java/com/spotify/styx/cli/CliExitException.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.cli;
+
+public class CliExitException extends RuntimeException {
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/CliExitException.java]: 🌟

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java [hunk 2]
```diff
@@ -282,8 +290,7 @@ private void workflowDelete() throws ExecutionException, InterruptedException {
               + workflows.stream().collect(joining(", "))
               + "in component " + component + "? [yN]").trim();
       if (!reply.equals("y")) {
-        System.exit(1);
-        return;
+        throw new CliExitException(1);
```
Comment (bergman) [inline styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: use the constants please

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java [hunk 3]
```diff
@@ -784,4 +791,38 @@ public boolean consumeArgument() {
       return true;
     }
   }
+
+  interface CliContext {
+
+    CliOutput output(OUTPUT output);
+
+    enum OUTPUT {
```
Comment (bergman) [inline styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: shouldn't the type name be cased like regular classes/types and only the values UPPERCASE?

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java (hunk 2 above)
Comment (danielnorberg) [reply styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: 👍

File: styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java (hunk 3 above)
Comment (danielnorberg) [reply styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java]: yeah this code was written very quickly 😂

**Comment by bergman** [comment]
👍

**Pull request merged**